backends-common/postgres/src/test/java/org/apache/james/backends/postgres/SimpleJamesPostgresConnectionFactoryTest.java

```diff
@@ -0,0 +1,100 @@
+
+package org.apache.james.backends.postgres;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.backends.postgres.utils.JamesPostgresConnectionFactory;
+import org.apache.james.backends.postgres.utils.SimpleJamesPostgresConnectionFactory;
+import org.apache.james.core.Domain;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import io.r2dbc.postgresql.PostgresqlConnectionConfiguration;
+import io.r2dbc.postgresql.PostgresqlConnectionFactory;
+import io.r2dbc.spi.Connection;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Testcontainers
+public class SimpleJamesPostgresConnectionFactoryTest extends JamesPostgresConnectionFactoryTest {
+    static final String DB_NAME = "james-db";
+    static final String DB_USER = "james";
+    static final String DB_PASSWORD = "1";
+
+    @Container
+    static final GenericContainer<?> container = new PostgreSQLContainer("postgres:11.1")
+        .withDatabaseName(DB_NAME)
+        .withUsername(DB_USER)
+        .withPassword(DB_PASSWORD);
+
+    private PostgresqlConnectionFactory connectionFactory;
+    private SimpleJamesPostgresConnectionFactory jamesPostgresConnectionFactory;
+
+    JamesPostgresConnectionFactory jamesPostgresConnectionFactory() {
+        return jamesPostgresConnectionFactory;
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        container.stop();
+        container.start();
+        connectionFactory = new PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder()
+            .host(container.getHost())
+            .port(container.getMappedPort(5432))
+            .username(DB_USER)
+            .password(DB_PASSWORD)
+            .database(DB_NAME)
+            .build());
+        jamesPostgresConnectionFactory = new SimpleJamesPostgresConnectionFactory(connectionFactory);
+    }
+
+    @Test
+    void factoryShouldCreateCorrectNumberOfConnection() {
+        // create 50 connection
+        Flux.range(1, 50)
+            .flatMap(i -> jamesPostgresConnectionFactory.getConnection(Domain.of("james"+i)))
+            .last()
+            .block();
+
+        Integer dbActiveConnectionNumber = Mono.from(connectionFactory.create())
+            .flatMap(connection -> Mono.from(connection.createStatement("SELECT count(*) from pg_stat_activity where usename = $1;")
+                .bind("$1", DB_USER)
+                .execute()))
+            .flatMap(result -> Mono.from(result.map((row, rowMetadata) -> row.get(0, Integer.class))))
+            .block();
+
+        // It should be 51 connections to the DB plus one connection to check number of connection
+        assertThat(dbActiveConnectionNumber).isEqualTo(52);
```

Comment:
We create 50 connections, so why it should be 51 connections (?) + 1? Would expect 50 + 1?

Reply:
fixed